cpp/ql/test/query-tests/Security/CWE/CWE-732/OpenCallMissingModeArgument.c

```diff
@@ -0,0 +1,16 @@
+typedef unsigned int mode_t;
+
+#define O_CREAT 0100
+
+int open(const char *pathname, int flags, ...);
+
+int openat(int dirfd, const char *pathname, int flags, ...);
```

Comment:
Yeah, I actually started out with 4 prototypes (two for `open` and two for `openat`), as you might expect, but of course there's no overloading in C, so that's not possible. So I started looking at actual implementations, and this seems to way this is done in C standard libraries. See, e.g., [OpenBSD](https://github.com/openbsd/src/blob/2207c4325726fdc5c4bcd0011af0fdf7d3dab137/sys/sys/fcntl.h#L208-L216).

Alternative would be to switch to C++.

Reply:
Ah, I hadn't thought of that.  If its the way OpenBSD does it, I'm happy.

Reply:
Neither had I. It does explain why you can get random bytes from the stack though, as that's what variadic functions do when you don't provide enough arguments.